base/test.jl

```diff
@@ -22,6 +22,30 @@ export GenericString
 
 #-----------------------------------------------------------------------
 
+# Backtrace utility functions
+function ip_matches_name(ip, dir::String, file::String)
+    for fr in StackTraces.lookup(ip)
+        if fr === StackTraces.UNKNOWN || fr.from_c
+            return false
+        end
+        path = string(fr.file)
+        dirname(path) == dir && basename(path) == file && return true
+    end
+    return false
+end
+
+function scrub_backtrace(bt)
+    do_test_ind = findfirst(addr->Base.REPL.ip_matches_func(addr, :do_test), bt)
```

Comment:
Can you more specifically explain the issue you have. Is it,  "Assuming that the `do_test` gets inlined (it won't) and then julia is compiled in such a way that the inline frame dissapears from the backtrace, then a user that have a function called `do_test` (that is not removed) will be removed in the filtering"?